src/features/comments.ts

```diff
@@ -0,0 +1,115 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { createComment, deleteComment, getPostComments } from '../api/comments';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+
+type CommentsState = {
+  comments: Comment[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: CommentsState = {
+  comments: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const fetchComments = createAsyncThunk<Comment[], Post['id']>(
+  'comments/fetch',
+  async (postId, { rejectWithValue }) => {
+    try {
+      return await getPostComments(postId);
+    } catch (error) {
+      return rejectWithValue(
+        (error as Error).message || 'Failed to fetch comments',
+      );
+    }
+  },
+);
+
+export const addComment = createAsyncThunk<Comment, Omit<Comment, 'id'>>(
+  'comments/add',
+  async (comment, { rejectWithValue }) => {
+    try {
+      return await createComment(comment);
+    } catch (error) {
+      return rejectWithValue(
+        (error as Error).message || 'Failed to add comment',
+      );
```

Comment:
Same as my previous comment, you should add a new property to the state to hold the error message, and set it here. This way, you can display the error message to the user without losing the state of the comments.